packages/perseus/src/widgets/passage/passage-markdown.jsx

```diff
@@ -347,70 +345,90 @@ const rules: $FlowFixMe = {
                 content: capture[1],
             };
         },
-        react: (node, output, state) => {
+        react: (node: HighlightNode) => {
             return [
                 <span key={0} className="perseus-review-highlight">
                     {node.content}
                 </span>,
             ];
         },
     },
-    // $FlowFixMe[prop-missing]
-    // $FlowFixMe[incompatible-use]
     strong: SimpleMarkdown.defaultRules.strong,
-    // $FlowFixMe[prop-missing]
-    // $FlowFixMe[incompatible-use]
     u: SimpleMarkdown.defaultRules.u,
-    // $FlowFixMe[prop-missing]
-    // $FlowFixMe[incompatible-use]
     em: SimpleMarkdown.defaultRules.em,
-    // $FlowFixMe[prop-missing]
-    // $FlowFixMe[incompatible-use]
     del: SimpleMarkdown.defaultRules.del,
     text: {
-        // $FlowFixMe[prop-missing]
-        // $FlowFixMe[incompatible-use]
         ...SimpleMarkdown.defaultRules.text,
         react(node, output, state) {
             return <span key={state.key}>{node.content}</span>;
         },
     },
 };
 
-const INITIAL_PARSE_STATE = {
-    currentRef: [],
-    useRefs: true,
-    lastRef: 0,
-    lastFootnote: {id: 0, text: ""},
+const START_REF_PREFIX = "start-ref-";
+const END_REF_PREFIX = "end-ref-";
+const REF_STYLE = {
+    display: "inline-block",
+    width: 0,
+    visibility: "hidden",
 };
-// $FlowFixMe[prop-missing]
+
+const LABEL_OUTER_STYLE = {
+    // for some reason we need these to keep the nbsp from wrapping when the
+    // inner circle/square is display: inline-block
+    display: "inline",
+    whiteSpace: "nowrap",
+};
+
+const SQUARE_LABEL_STYLE = {
+    display: "inline-block",
+    color: "rgb(255, 255, 255)",
+    backgroundColor: "rgb(90, 90, 90)",
+    paddingLeft: 10,
+    paddingRight: 10,
+    userSelect: "none",
+    WebkitUserSelect: "none",
+};
+
+const CIRCLE_LABEL_STYLE = {
+    display: "inline-block",
+    color: "rgb(255, 255, 255)",
+    backgroundColor: "rgb(90, 90, 90)",
+    userSelect: "none",
+    WebkitUserSelect: "none",
+    width: 22,
+    height: 22,
+    borderRadius: "50%",
+    textAlign: "center",
+};
+
 const builtParser = SimpleMarkdown.parserFor(rules);
-const parse: (string, $FlowFixMe) => $FlowFixMe = (source, state) => {
+const parse: (string, ?ParseState) => Array<SingleASTNode> = (
+    source,
+    state,
+) => {
     state = state || {};
     const paragraphedSource = source + "\n\n";
     return builtParser(paragraphedSource, _.extend(state, INITIAL_PARSE_STATE));
 };
 
+const output: (Array<SingleASTNode>) => React.Node = SimpleMarkdown.reactFor(
+    SimpleMarkdown.ruleOutput(rules, "react"),
+);
+
 export default {
     parse: parse,
-    // $FlowFixMe[prop-missing]
-    output: (SimpleMarkdown.reactFor(
-        // $FlowFixMe[prop-missing]
-        SimpleMarkdown.ruleOutput(rules, "react"),
-    ): $FlowFixMe),
-    START_REF_PREFIX: START_REF_PREFIX,
-    END_REF_PREFIX: END_REF_PREFIX,
-    _rulesForTesting: rules,
+    output: output,
```

Comment:
You could also simplify these (I can't remember what the JS term is for this):
```suggestion
    parse,
    output,
```